server/datastore/mysql/apple_mdm.go

```diff
@@ -2181,6 +2182,57 @@ func (ds *Datastore) InsertMDMAppleBootstrapPackage(ctx context.Context, bp *fle
 	return nil
 }
 
+func (ds *Datastore) CopyDefaultMDMAppleBootstrapPackage(ctx context.Context, ac *fleet.AppConfig, toTeamID uint) error {
+	if ac == nil {
+		return ctxerr.New(ctx, "app config must not be nil")
+	}
+	if toTeamID == 0 {
+		return ctxerr.New(ctx, "team id must not be zero")
+	}
+
+	return ds.withRetryTxx(ctx, func(tx sqlx.ExtContext) error {
+		// Copy the bytes for the default bootstrap package to the specified team
+		insertStmt := `
+INSERT INTO mdm_apple_bootstrap_packages (team_id, name, sha256, bytes, token)
+SELECT ?, name, sha256, bytes, ?
+FROM mdm_apple_bootstrap_packages
+WHERE team_id = 0
+`
+		_, err := tx.ExecContext(ctx, insertStmt, toTeamID, uuid.New().String())
+		if err != nil {
+			if isDuplicate(err) {
+				return ctxerr.Wrap(ctx, &existsError{
+					ResourceType: "BootstrapPackage",
+					TeamID:       &toTeamID,
+				})
+			}
+			return ctxerr.Wrap(ctx, err, fmt.Sprintf("copy default bootstrap package to team %d", toTeamID))
+		}
+
+		// Update the team config json with the default bootstrap package url
+		//
+		// NOTE: The bytes copied above may not match the bytes at the url because it is possible to
+		// upload a new bootrap pacakge via the UI, which replaces the bytes but does not change
+		// the configured URL. This was a deliberate product design choice and it is intended that
+		// the bytes would be replaced again the next time the team config is applied (i.e. via
+		// fleetctl in a gitops workflow).
+		url := ac.MDM.MacOSSetup.BootstrapPackage.Value
+		if url != "" {
+			updateConfigStmt := `
+UPDATE teams 
+SET config = JSON_MERGE_PATCH(config, '{ "mdm": { "macos_setup": { "bootstrap_package": "%s" } } }') 
+WHERE id = ?
```

Comment:
`JSON_MERGE_PATCH` was added in MySQL 5.7.22. We test against 5.7.21 in [CI](https://github.com/fleetdm/fleet/actions/runs/5733019996/job/15537000835?pr=13089#step:9:8270). Do we have a hard cutoff at 5.7.21 or can we bump the tests to 5.7.22?

Reply:
unfortunately (because this is really neat) I don't think we can do it, mainly because we document `5.7.21` as the supported version in [our FAQ](https://fleetdm.com/docs/get-started/faq#what-mysql-versions-are-supported).

seems like we didn't document _why_ we choose that specific version, but if I have to guess it must be the default used somewhere (like a cloud provider)

Reply:
can we use [JSON_MERGE](https://dev.mysql.com/doc/refman/5.7/en/json-modification-functions.html#function_json-merge) instead?

Reply:
No, because `bootstrap_package` exists in the newly-created team config (albeit `null`), `JSON_MERGE` will append the old and new values as `[null, "https://example.com/bs.pkg"]`.  `JSON_SET` works, I think.